source/material_model/utilities.cc

```diff
@@ -699,6 +699,43 @@ namespace aspect
       }
 
 
+      // Calculate a parameter of single composition at single point by averaging phases.
+      // Either linear(by_log is 0) or using log form(by_log is 1), default is linear.
+      double phase_average_value (const std::pair<std::vector<double>, const std::vector<unsigned int>> &gamma_inputs,
+                                  const std::vector<double> &parameter_values,
+                                  const unsigned int composition,
+                                  bool by_log)
+      {
+        const std::vector<double> &gamma_values = gamma_inputs.first;
+        const std::vector<unsigned int> &n_phases_per_composition = gamma_inputs.second;
+        // Calculate base index and assign base value
+        unsigned int base = 0;
+        for (unsigned int i=0; i<composition; ++i)
+          base += n_phases_per_composition[i] + 1;
+        double averaged_parameter = parameter_values[base];
+        if (n_phases_per_composition[composition] > 0)
+          {
+            // Do averaging when there are multiple phases
+            if (by_log)
+              averaged_parameter = log(averaged_parameter);
+            for (unsigned int i=0; i<n_phases_per_composition[composition]; ++i)
+              {
+                Assert(base+i+1<parameter_values.size(), ExcInternalError());
+                if (by_log)
+                  {
+                    // First average by log values and then take the exponential.
+                    // This is used for averaging prefactors in flow laws.
+                    averaged_parameter += gamma_values[base-composition+i] * log(parameter_values[base+i+1] / parameter_values[base+i]);
+                  }
+                else
```

Comment:
```suggestion
                else if (operation == MaterialAveraging::arithmetic_average)
                   averaged_parameter += gamma_values[base-composition+i] * (parameter_values[base+i+1] - parameter_values[base+i]);
                else
                  AssertThrow(false, ExcInternalError());
```

Reply:
@gassmoeller Hi, Rene, I think I have addressed all of your comments, with expect that I have to name the variable ar phase_transition_arithmetic in PhaseAveragingOperation in order to not conflict with the one defined in CompositionalAveragingOperation, is there a better solution to that?

Reply:
And right now, I see the first checking failed with the cookbook and some issues with make which I don't quite understand.